dffml/secret/file.py

```diff
@@ -0,0 +1,46 @@
+import os
+import hashlib
+import asyncio
+import pathlib
+from typing import Union,Any
+
+from ..base import config
+from ..util.entrypoint import entrypoint
+from .base import BaseSecretContext, BaseSecret
+
+from dffml.high_level import Record
+from ..source.memory import MemorySourceContext
+from ..source.ini import INISourceConfig,INISource
+
+
+
+class FileSecretContext(BaseSecretContext, MemorySourceContext):
+    def __init__(self,*args,**kwargs):
+        super().__init__(*args,**kwargs)
+
+
+    async def get(self, name: str) -> Union[None, str]:
+        with open(self.parent.config.filename,'r+') as fd:
+            await self.parent.load_fd(fd)
+        record = self.parent.mem.get(name,None)
+        if not record:
+            return None
+        return record.feature('data')
+
+    async def set(self, name: str, value: Any):
+        record = Record(name, data={"features": {"data": value}})
+        await self.update(record)
+        with open(self.parent.config.filename,'w+') as fd:
+            await self.parent.dump_fd(fd)
+
+
+
+@entrypoint("file")
+class FileSecret(BaseSecret,INISource):
+
+    CONTEXT = FileSecretContext
+    CONFIG = INISourceConfig
+
+    def __init__(self,*args,**kwargs):
+        super().__init__(*args,**kwargs)
+        self.mem: Dict[str, Record] = {}
```

Comment:
We should also probably issue a warning about this being insecure in this type of style (with the extra newlines so people are sure to see it)

```python
        warnings.warn(
            """


            INSECURE This really does pbkdf2_hmac and not scrypt (since that
            requires openssl 1.1 and most systems only have 1.0.x at time of
            writing) Also its just the example I copy pasted from the docs to
            illustrate threading. 100000 is probably not enough iterations!!!

            """
        )
```

Reply:
> We should probably put mem in the `Context` since the parent isn't going to use it

We are using `update` and `load_fd` which expects `parent` to have `mem`. Am I missing something here?